lib/ui/src/components/sidebar/utils.ts

```diff
@@ -0,0 +1,88 @@
+import memoize from 'memoizerific';
+import { document, window, DOCS_MODE } from 'global';
+import { SyntheticEvent } from 'react';
+import { StoriesHash, isRoot } from '@storybook/api';
+
+import { DEFAULT_REF_ID } from './data';
+import { Item, RefType, Dataset, SearchItem } from './types';
+
+export const storyLink = (storyId: string, refId: string) => {
+  const type = DOCS_MODE ? 'docs' : 'story';
+  const refPrefix = refId === DEFAULT_REF_ID ? '' : `${refId}_`;
+  return `${document.location.pathname}?path=/${type}/${refPrefix}${storyId}`;
+};
+
+export const prevent = (e: SyntheticEvent) => {
+  e.preventDefault();
+  return false;
+};
+
+export const createId = (id: string, prefix: string) => `${prefix}_${id}`;
+
+export const get = memoize(1000)((id: string, dataset: Dataset) => dataset[id]);
```

Comment:
do these memos actually improve perf?

Reply:
Not sure. They were there before: https://github.com/storybookjs/storybook/blob/next/lib/ui/src/components/sidebar/Tree/utils.ts#L58

Reply:
In the newly added places they are needed to prevent unnecessary rerenders I believe.